pkg/util/config/config.go

```diff
@@ -74,7 +74,15 @@ func (m *Mux) Channel(source string) chan interface{} {
 	}
 	newChannel := make(chan interface{})
 	m.sources[source] = newChannel
-	go wait.Until(func() { m.listen(source, newChannel) }, 0, wait.NeverStop)
+
+	var stopCh <-chan struct{}
+	if len(stopChes) > 0 {
+		stopCh = stopChes[0]
+	} else {
+		stopCh = wait.NeverStop
+	}
```

Comment:
Rather than doing this for backwards compatibility, I wonder if it'd be cleaner to add a new method to the type. E.g. `ChannelWithContext(ctx context.Context, source string) chan interface{}`.

Reply:
Actually, this is not in library code (not in `staging`) so might just break the signature and update the callers.

Reply:
hi @ash2k,  if we add a new method there, do we need to look at all callers?  If so, I would like the PR to add the new method and update `TestConfigurationChannels `. Then create a new PR to update all the callers. and decide whether they should call the new method or the old.

Reply:
I think we don't need yet another method, we can just change this one. The caller should always be responsible for stopping the goroutines they start, so a method without a way to stop them is a bad API. Hence, in my opinion, we should change this one rather just than adding a new one (change bad API to good API).

Reply:
I found a lot of code in the kubelet that uses wait.NeverStop, which doesn't seem to care about stopping the goroutine. If we change this method to `ChannelWithContext(ctx context.Context, source string) chan interface`, we need to cancel this ctx somewhere uniformly, but I can't find the right place

Reply:
It's fine, we don't need to fix the world in this PR :)